core/src/primitives/rectangle/mod.rs

```diff
@@ -246,6 +246,65 @@ impl Rectangle {
         Rectangle::zero()
     }
 
+    /// Returns the minimum sized `Rectangle` that encompasses both `self` and `other`.
+    ///
+    /// # Examples
+    ///
+    /// ## Encompass
+    ///
+    /// This example draws two rectangles to a mock display using the `.` character, along with
+    /// their encompassing rectangle shown with `#` characters.
+    ///
+    /// ```rust
+    /// use embedded_graphics::{
+    ///     mock_display::MockDisplay, pixelcolor::BinaryColor, prelude::*,
+    ///     primitives::{Rectangle, PrimitiveStyle},
+    /// };
+    ///
+    /// let mut display = MockDisplay::new();
+    /// # display.set_allow_overdraw(true);
+    ///
+    /// let rect1 = Rectangle::new(Point::zero(), Size::new(7, 8));
+    /// let rect2 = Rectangle::new(Point::new(2, 3), Size::new(10, 7));
+    ///
+    /// let encompass = rect1.encompass(&rect2);
+    ///
+    /// rect1
+    ///     .into_styled(PrimitiveStyle::with_stroke(BinaryColor::Off, 1))
+    ///     .draw(&mut display)?;
+    ///
+    /// rect2
+    ///     .into_styled(PrimitiveStyle::with_stroke(BinaryColor::Off, 1))
+    ///     .draw(&mut display)?;
+    ///
+    /// encompass
+    ///     .into_styled(PrimitiveStyle::with_stroke(BinaryColor::On, 1))
+    ///     .draw(&mut display)?;
+    ///
+    /// display.assert_pattern(&[
+    ///     "############",
+    ///     "#     .    #",
+    ///     "#     .    #",
+    ///     "# .........#",
+    ///     "# .   .    #",
+    ///     "# .   .    #",
+    ///     "# .   .    #",
+    ///     "#......    #",
+    ///     "# .        #",
+    ///     "############",
+    /// ]);
+    /// # Ok::<(), core::convert::Infallible>(())
+    /// ```
+    pub fn encompass(&self, other: &Rectangle) -> Rectangle {
+        let top_left = self.top_left.component_min(other.top_left);
+        let bottom_right = self
+            .bottom_right()
+            .unwrap_or(self.top_left)
+            .component_max(other.bottom_right().unwrap_or(other.top_left));
```

Comment:
```suggestion
        let bottom_right = self.anchor_point(AnchorPoint::BottomRight)
            .component_max(other.anchor_point(AnchorPoint::BottomRight));
```

Reply:
Good example of why it's probably best if e-g provides such a method directly 😄